### PR TITLE
[JENKINS-67962]The giant "No builds" message is shown even with a pending build

### PR DESCRIPTION
**Pull request by Langer0416**
See [JENKINS-67962](https://issues.jenkins-ci.org/browse/JENKINS-67962).
before: If your job is in the pending stage, it will show the "No builds" message even if you refresh the page.

before

after: The job in the pending stage won't show the "No builds" message.

after

### Proposed changelog entries

* Hide "No builds" when there is a pending build.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
My testing results look like your explanation of the current behavior:

https://user-images.githubusercontent.com/13383509/162734276-2f7c38c1-dbeb-4de2-8f8e-9d143a0b0a95.mp4

How did you test it to provide the presented results?

**Comment by NotMyFault** [comment]
> @NotMyFault Reload the page once the build is queued. See my comment in Jira.

Oic, thanks for the clarification.

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
